test/scheduler/suite_test.go

```diff
@@ -0,0 +1,382 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package tests features a number of test suites that verify the behavior of the scheduler
+// and its related components.
+package tests
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap/zapcore"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler"
+	"go.goms.io/fleet/pkg/scheduler/clustereligibilitychecker"
+	"go.goms.io/fleet/pkg/scheduler/queue"
+	"go.goms.io/fleet/pkg/scheduler/watchers/clusterresourceplacement"
+	"go.goms.io/fleet/pkg/scheduler/watchers/clusterschedulingpolicysnapshot"
+	"go.goms.io/fleet/pkg/scheduler/watchers/membercluster"
+)
+
+const (
+	defaultProfileName   = "defaultProfile"
+	defaultSchedulerName = "defaultScheduler"
+
+	customDeletionBlockerFinalizer = "custom-deletion-blocker"
+
+	dummyReason = "dummyReason"
+
+	eventuallyDuration   = time.Second * 5
+	eventuallyInterval   = time.Millisecond * 500
+	consistentlyDuration = time.Second * 1
+	consistentlyInterval = time.Millisecond * 200
+
+	memberCluster1  = "cluster-1-east-prod"
+	memberCluster2  = "cluster-2-east-prod"
+	memberCluster3  = "cluster-3-east-canary"
+	memberCluster4  = "cluster-4-central-prod"
+	memberCluster5  = "cluster-5-central-prod"
+	memberCluster6  = "cluster-6-west-prod"
+	memberCluster7  = "cluster-7-west-canary"
+	memberCluster8  = "unhealthy-cluster-east-prod"
+	memberCluster9  = "left-cluster-central-prod"
+	memberCluster10 = "non-existent-cluster"
```

Comment:
the real value of the cluster is not visible in the test cases. It's the name of the variable that really matters. In test case, I still see pick cluster1,2,3 instead verify cluster-1-east, cluster-2-east, cluster-3-east in the test case.